File: src/backend/controllers/employee.go
package controllers

import (
	"backend/models"

	beego "github.com/beego/beego/v2/server/web"
)

// Operations about journal
// /v1/journal
type EmployeeController struct {
	beego.Controller
}

// @Title GetAll
// @Description get all employees
// @Success 200 [] Employee
// @Failure 403 Bad request
// @Failure 500 Server error
// @router / [get]
func (e *EmployeeController) GetAll() {
	obs := models.GetEmployees()
	e.Data["json"] = obs
	e.ServeJSON()
}

// @Title GetById
// @Description get employee by id
// @Success 200 Employee
// @Failure 403 employee not found
// @Failure 500 Server error
//@router /:employeeId [get]
func (e *EmployeeController) GetById() {
	EmployeeId := e.Ctx.Input.Param(":employeeId")
	Employee, err := models.GetEmployee(EmployeeId)
	if err != nil {
		e.Data["json"] = err.Error()
	} else {
		e.Data["json"] = Employee
	}
	e.ServeJSON()
}

File: src/current-backend/controllers/employee.js
const Employee = require('../models/employee')

const queryEmployees = async (uid) => {
  try{
    const employees = await Employee.find({uid})
    return employees
  }catch(e){
    console.log(e)  
    throw e
  }
}



class EmployeeController{
  async getEmployees(req, res){
    try{
      const {user} = req 
      const employees = await queryEmployees(user._id)
      res.send(employees)
    }catch(e){
      console.log(e)
      res.status(400).json({error:e})
    }
  }
 
}

module.exports = new EmployeeController()